src/Kiota.Builder/Refiners/CommonLanguageRefiner.cs

```diff
@@ -396,7 +396,11 @@ protected static void ReplaceReservedNames(CodeElement current, IReservedNamesPr
             {
                 currentProperty.SerializationName = currentProperty.Name;
             }
-            current.Name = replacement.Invoke(current.Name);
+            var replacementName = replacement.Invoke(current.Name);
+            if (current.Parent is CodeClass parent
+                && parent?.GetChildElements().Select(x => x.Name.Equals(current.Name)) != null)
```

Comment:
parent is not going to be null here, you can remove the null propagation and the null check at the end.
Maybe the select should be a first or default instead?